src/JustSaying.Extensions.DependencyInjection.Microsoft/IServiceCollectionExtensions.cs

```diff
@@ -135,7 +140,11 @@ public static IServiceCollection AddJustSaying(this IServiceCollection services,
         services.TryAddSingleton<IMessageContextAccessor>(serviceProvider => serviceProvider.GetRequiredService<MessageContextAccessor>());
         services.TryAddSingleton<IMessageContextReader>(serviceProvider => serviceProvider.GetRequiredService<MessageContextAccessor>());
 
+#if NET8_0_OR_GREATER
+        services.TryAddSingleton<IMessageSerializationFactory>(sp => new TypedSystemTextJsonSerializationFactory(sp.GetRequiredService<IOptions<JsonSerializerOptions>>().Value));
```

Comment:
I'd suggest we add a custom holder type for this, then it's more intentional for the user where the serialization options are coming from, in case they want JS to use different defaults compared to _{some other thing they have}_.